week-1-ts-ecosystem/w01e04-light-me-up/src/PowerSource.ts

```diff
@@ -0,0 +1,11 @@
+export class PowerSource {
+  private energySupply = 100;
+
+  consume(energy) {
+    if (this.energySupply >= energy) {
+      this.energySupply -= energy;
+    } else {
+      throw new Error("Run out of energy");
+    }
```

Comment:
Jakbyś odwrócił warunek to uprościłoby zapis @NiestrojMateusz dlatego że wtedy `throw` przerwie i tak działanie metody.